src/slave/containerizer/mesos/paths.cpp

```diff
@@ -654,6 +654,30 @@ Option<ContainerID> parseCgroupPath(
   return current;
 }
 
+namespace cgroups2 {
+
+string agent(const string& root, bool leaf)
+{
+  if (leaf) { return path::join(root, "agent", "leaf"); }
+  else      { return path::join(root, "agent"); }
```

Comment:
IMO this looks nicer than using the ternary operator or removing the `else`, because the paths line up. Ditto below.